src/eos-metrics-instrumentation.c

```diff
@@ -284,6 +300,72 @@ record_live_boot (gpointer unused)
     return G_SOURCE_REMOVE;
 }
 
+static gchar *
+get_image_version_for_path (const gchar *path)
+{
+    ssize_t xattr_size;
+    gchar *image_version = NULL;
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR, NULL, 0);
+
+    if (xattr_size < 0)
+      return NULL;
+
+    image_version = g_malloc0 (xattr_size + 1);
+
+    xattr_size = getxattr (path, EOS_IMAGE_VERSION_XATTR,
+                           image_version, xattr_size);
+
+    /* this check is just in case the xattr has changed in between the
+     * size checks */
```

Comment:
Hmm, thinking about this again, we should be checking that `xattr_size` from the second call is equal to (or less than) `xattr_size` from the first call; as well as that it’s positive. Otherwise the xattr value could increase in size between the two calls, and we would end up with a truncated copy of it in `image_version`, and an `xattr_size` which is now longer than the allocated area for `image_version`.